src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -322,67 +311,31 @@ public void close() throws IOException {
     }
 
     @Override
-    public void stop() throws EngineException {
-        throw new UnsupportedOperationException("stop() is not supported by InternalEngine. Use InternalEngineHolder.");
-    }
-
-    @Override
     public ShardId shardId() {
```

Comment:
I think this can be removed (here and from the interface) and then `this.shardId` can be used in the only caller, the `.get()` method